pkg/vault/contracts/Vault.sol

```diff
@@ -885,13 +899,23 @@ contract Vault is IVaultMain, VaultCommon, Proxy {
         amountsOutRaw = new uint256[](locals.numTokens);
 
         for (uint256 i = 0; i < locals.numTokens; ++i) {
+            uint256 amountOutRaw;
+
             // 1) Calculate raw amount out.
-            // amountsOut are amounts exiting the Pool, so we round down.
-            // Do not mutate in place yet, as we need them scaled for the `onAfterRemoveLiquidity` hook
-            uint256 amountOutRaw = amountsOutScaled18[i].toRawUndoRateRoundDown(
-                poolData.decimalScalingFactors[i],
-                poolData.tokenRates[i]
-            );
+            {
+                uint256 amountOutScaled18 = amountsOutScaled18[i];
+                if (amountOutScaled18 > 0) {
+                    _ensureValidTradeAmount(amountOutScaled18);
+
+                    // amountsOut are amounts exiting the Pool, so we round down.
+                    // Do not mutate in place yet, as we need them scaled for the `onAfterRemoveLiquidity` hook
+                    amountOutRaw = amountsOutScaled18[i].toRawUndoRateRoundDown(
+                        poolData.decimalScalingFactors[i],
+                        poolData.tokenRates[i]
+                    );
+                }
+            }
```

Comment:
Do we need to also check `bptAmountIn`?

Reply:
I think so, yes. Will make sure, and add it.